src/qibo/models/error_mitigation.py

```diff
@@ -51,51 +51,82 @@ def get_gammas(noise_levels, analytical: bool = True):
     return zne_coefficients
 
 
-def get_noisy_circuit(circuit, num_insertions: int, insertion_gate: str = "CNOT"):
+def get_noisy_circuit(
+    circuit, num_insertions: int, global_unitary_folding=True, insertion_gate=None
+):
     """Standalone function to generate the noisy circuit with the inverse gate pairs insertions.
 
     Args:
         circuit (:class:`qibo.models.circuit.Circuit`): circuit to modify.
-        num_insertions (int): number of insertion gate pairs to add.
+        num_insertions (int): number of insertion gate pairs / global unitary folds to add.
+        global_unitary_folding (bool): Default True. Global unitary folding is carried out by default.
+            Global unitary folding takes precedence over insertion_gate.
         insertion_gate (str, optional): gate to be used in the insertion.
             If ``"RX"``, the gate used is :math:``RX(\\pi / 2)``.
```

Comment:
```suggestion
        num_insertions (int): number of insertion gate pairs / global unitary folds to add.
        global_unitary_folding (bool): If ``True``, noise is increased by global unitary folding.
            If ``False``, local unitary folding is used. Defaults to ``True``.
        insertion_gate (str, optional): gate to be folded in local unitary folding.
            If ``RX``, the gate used is :math:`RX(\\pi / 2)`. Otherwise, it is the ``CNOT`` gate.
```